furnace.c

```diff
@@ -46,6 +46,16 @@ typedef struct {
   int             last_temp;
 } pilot_context_t;
 
+#if CONFIG_PWM_MAPPER
+typedef struct {
+  absolute_time_t mapper_deadline;
```

Comment:
It's already part of `mapper_context_t`. Consider renaming to just `absolute_time_t deadline`.